tutorials/en/vectara-beginner-app-tutorial-showcase-the-creation-of-vectara-app-in-legal-use-case.mdx

```diff
@@ -0,0 +1,390 @@
+
+### Langchain 🤖
+Langchain, the engine of text generation in the application, sifts through the legal insights extracted by Vectara to generate automated legal advice. It is the cornerstone that enables the application to furnish text-based consultations, making legal advice easily accessible.
+
+The architecture of the application is elegantly simple yet potent. The user-friendly interface, sculpted with Streamlit, facilitates effortless uploading of PDF documents. Upon upload, Vectara springs into action, processing the document to extract legal insights. Langchain then takes the baton, generating legal advice that is promptly displayed to the user. This synergy ensures that users not only receive instant legal consultations but also have the option for further discussions with legal experts if needed.
+
+Dive into this tutorial to traverse the developmental voyage of this Legal Consultation Application and explore the plethora of features awaiting your discovery!
+
+---
+
+### Setting the Stage: Setup and Installation Guide 🛠️
+
+Before we delve into the realms of code and explore the intricacies of our application, it's imperative to set the stage right. This segment is dedicated to guiding you through the process of setting up and installing the necessary components for our application. The emphasis is on ensuring a smooth sail as we venture into the development phase.
+
+#### Step 1: Create a Virtual Environment
+Creating a virtual environment is a good practice to manage dependencies and ensure the application runs consistently across different setups.
+
+```bash
+python3 -m venv vectara-env
+```
+
+Activate the virtual environment:
+- On Windows:
+```bash
+.\vectara-env\Scripts\activate
+```
+- On macOS and Linux:
+```bash
+source vectara-env/bin/activate
+```
+
+#### Step 2: Install Necessary Packages
+Install the necessary packages using pip:
+
+```bash
+pip install streamlit python-dotenv langchain 
+```
+
+#### Step 3: Create the .env File
+Create a file named `.env` in the root directory of your project. This file will store your environment variables. Here's how your `.env` file should look like, replace the placeholders with your actual credentials:
+
+```plaintext
+CUSTOMER_ID=your-customer-id
+CORPUS_ID=your-corpus-id
+API_KEY=your-api-key
+OPENAI_API_KEY=openai_api_key here
```

Comment:
This is one of the issues of using LangChain in this way - you have to have an OpenAI key, where participants in the hackathon may not have it. Maybe we can add a comment saying this does not have to be OpenAI, and could be any model, including Llama2 which will be available to participants via Anyscale Endpoints?